docs/workflows/create_workflow_block.md

```diff
@@ -0,0 +1,1898 @@
+            ImageParentMetadata,
+            Batch,
+        )
+        from inference.core.workflows.execution_engine.entities.types import (
+            BATCH_OF_IMAGES_KIND,
+            BATCH_OF_OBJECT_DETECTION_PREDICTION_KIND,
+            StepOutputImageSelector,
+            StepOutputSelector,
+            WorkflowImageSelector,
+        )
+        from inference.core.workflows.prototypes.block import (
+            BlockResult,
+            WorkflowBlock,
+            WorkflowBlockManifest,
+        )
+        
+        class BlockManifest(WorkflowBlockManifest):
+            type: Literal["my_block/dynamic_crop@v1"]
+            image: Union[WorkflowImageSelector, StepOutputImageSelector]
+            predictions: StepOutputSelector(
+                kind=[BATCH_OF_OBJECT_DETECTION_PREDICTION_KIND],
+            )
+
+            @classmethod
+            def accepts_batch_input(cls) -> bool:
+                return True
+        
+            @classmethod
+            def get_output_dimensionality_offset(cls) -> int:
+                return 1
+        
+            @classmethod
+            def describe_outputs(cls) -> List[OutputDefinition]:
+                return [
+                    OutputDefinition(name="crops", kind=[BATCH_OF_IMAGES_KIND]),
+                ]
+        
+            @classmethod
+            def get_execution_engine_compatibility(cls) -> Optional[str]:
+                return ">=1.0.0,<2.0.0"
+
+        class DynamicCropBlockV1(WorkflowBlock):
+
+            @classmethod
+            def get_manifest(cls) -> Type[WorkflowBlockManifest]:
+                return BlockManifest
+            
+            def run(
+                self,
+                image: Batch[WorkflowImageData],
+                predictions: Batch[sv.Detections],
+            ) -> BlockResult:
+                results = []
+                for single_image, detections in zip(image, predictions):
+                    crops = []
+                    for (x_min, y_min, x_max, y_max) in detections.xyxy.round().astype(dtype=int):
+                        cropped_image = single_image.numpy_image[y_min:y_max, x_min:x_max]
+                        parent_metadata = ImageParentMetadata(parent_id=f"{uuid4()}")
+                        if cropped_image.size:
+                            result = WorkflowImageData(
+                                parent_metadata=parent_metadata,
+                                numpy_image=cropped_image,
+                            )
+                        else:
+                            result = None
+                        crops.append({"crops": result})
+                    results.append(crops)
+                return results
+        ```
+      
+        * in lines `31-33` manifest declares that block accepts batches of inputs
+
+        * in lines `35-37` manifest class declares output dimensionality 
+        offset - value `1` should be understand as adding `1` to dimensionality level
```

Comment:
`should be understand` -> `should be understood`

Reply:
👍